src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -136,7 +139,13 @@ class NotebookEditor extends Component {
     window.addEventListener('beforeunload', this.beforeUnload)
 
     try {
-      const { clusterName, clusterUrl } = await this.startCluster()
+      const { clusterName, clusterUrl, error } = await this.startCluster()
+
+      if (error) {
+        this.setState({ clusterError: error, failed: true })
+        return
```

Comment:
nit: since this is an error case, it might be slightly clearer if it was handled via a rejection with a special value